Document the ClearlyDefined certifier

Ritesh-Udgata opened the pull request:
Closes #131

funnelfiasco commented on guac-clearly-defined-certifier.md:
```diff
@@ -0,0 +1,37 @@
+## Documentation: GUAC Pull from ClearlyDefined
```
Let's make a few changes here. First, you don't need to say "documentation" in the header because it's on the docs site already. Second, you can replace this line with something like the below so that it will render in the CI build:

```
---
layout: page
title: ClearlyDefined certifier
permalink: /certifier-clearlydefined/
---
```

funnelfiasco commented on guac-clearly-defined-certifier.md:
```diff
@@ -0,0 +1,37 @@
+
+### Overview
+
+GUAC (Graph for Understanding Artifact Composition) integrates with **ClearlyDefined** to enhance supply chain transparency by retrieving accurate license data for software dependencies. This functionality helps organizations make informed decisions about software licenses when managing their dependencies.
```
Link to clearlydefined.io

funnelfiasco commented on guac-clearly-defined-certifier.md:
```diff
@@ -0,0 +1,37 @@
+
+This setup allows GUAC to remain flexible while keeping license data as accurate and complete as possible.
+
+For further details and potential enhancements, such as batched query support, you can explore the [official GUAC documentation](https://docs.guac.sh) or related updates from the development team.
```
This site is the official docs. :-) Let's just remove this line.

funnelfiasco commented on guac-clearly-defined-certifier.md:
```diff
@@ -0,0 +1,37 @@
+GUAC integrates ClearlyDefined queries either by:
+- **Scheduled certifier execution**: Automatically runs at set intervals to keep data current.
+- **On-demand during SBOM ingestion**: Queries ClearlyDefined in real-time during dependency ingestion, with the trade-off of slower processing.
+
```
There's a third option: running `guacone certifier cd` from the command line

funnelfiasco reviewed (changes requested):
This is a great start. Apart from the comments I left, there are two other things that I'd like to see:

1. A link to the [configuration guide](https://docs.guac.sh/guac-configuration/#certifier-configuration), perhaps in the invocation section when talking about running automatically.
2. A table of the supported command line arguments and what they mean. See PR #165 for an example

Ritesh-Udgata commented:
1. Added link to configuration guide
2. Added table of supported command line argument.
3. Changed Title
4. Link to clearlydefined.io
5. Removed official doc link
6. Added third option to integrate ClearlyDefined into Guac
7. Formatted document using Prettier
8. committed along with signoff and gpg-sign

funnelfiasco commented on guac-clearly-defined-certifier.md:
````diff
@@ -0,0 +1,74 @@
+3. **Manual Command Line Execution**  
+   - Users can directly invoke the certifier using:  
+     ```bash
+     guacone certifier cd --input <path> --output <path> [options]
````
The `--input` and `--output` arguments here are not supported by the `guacone certifier` command

Ritesh-Udgata replied:
Since I am not sure about alternatives as guacone certifier may have specific arguments for choosing between source types rather than directly setting files via --input and --output, I think I will remove the command and only keep the description.

funnelfiasco replied:
The help content I included in another thread should give you the guidance you need here.

funnelfiasco commented on guac-clearly-defined-certifier.md:
```diff
@@ -0,0 +1,74 @@
+
+---
+
+Below is a table of the supported **command-line arguments** for the **ClearlyDefined certifier**. These flags allow users to control input, output, and execution behavior. 
```
Where did you get this list of arguments, because it does not match the output of `guacone certifier cd --help`?

Ritesh-Udgata replied:
could you please provide the ouputs of guacone certifier cd --help. 
I went through the blogs and code base of certifier but couldn't figure out a relevant change.

funnelfiasco replied:
```bash
bcotton@Bens-MacBook-Air ~/g/guac (main)> guacone certifier cd --help
runs the clearly defined certifier

Usage:
  guacone certifier cd [flags]

Flags:
      --certifier-batch-size int   sets the batch size for pagination query for the certifier (default 60000)
      --certifier-latency string   sets artificial latency on the certifier. Defaults to empty string (not enabled) but can set m, h, s...etc
  -h, --help                       help for cd
  -l, --last-scan int              hours since the last scan was run. If not set, run on all packages/sources (default 4)

Global Flags:
      --add-license-on-ingest   if enabled, the ingestor will query and ingest clearly defined for licenses. Warning: This will increase ingestion times
      --add-vuln-on-ingest      if enabled, the ingestor will query and ingest OSV for vulnerabilities. Warning: This will increase ingestion times
      --csub-addr string        address to connect to collect-sub service (default "localhost:2782")
      --csub-tls                enable tls connection to the server
      --csub-tls-skip-verify    skip verifying server certificate (for self-signed certificates for example)
      --gql-addr string         endpoint used to connect to graphQL server (default "http://localhost:8080/query")
      --header-file string      a text file containing HTTP headers to send to the GQL server, in RFC 822 format
  -i, --interval string         if polling set interval, m, h, s, etc. (default "5m")
  -p, --poll                    sets the collector or certifier to polling mode
```

Ritesh-Udgata commented:
DONE 👍

funnelfiasco reviewed (approved):
LGTM, thanks!

funnelfiasco reviewed (changes requested):
I spoke too soon. Two things you need to do:

1. Run prettier (see README.md) to fix the formatting
2. Squash the commits into one.

Otherwise, it's good to go.

Ritesh-Udgata commented:
1. Ran through Prettier. Came out exactly the same.
2. Squashed all commits into the initial commit. Kept the commit message same as initial commit. 
3. Added sign off and gpg-key
4. force pushed the branch

funnelfiasco commented:
> Ran through Prettier. Came out exactly the same.

It shouldn't have. Prettier will wrap lines at 80 characters if you use `npx --yes prettier --write --prose-wrap always *.md`, which is what the CI build (which just failed) does. Looking at your file, there are still several lines longer than 80 characters

funnelfiasco reviewed (approved):
Looks good, thanks @Ritesh-Udgata!

Ritesh-Udgata commented:
Will I be receiving a swag for the contributions ?

funnelfiasco commented:
You only need to ask once. I'll send the form to the email address you used for your commit.
